### ISSUE TITLE
"tags" always updated if unsorted or are comma-separated

### ISSUE THREAD

**Issue by needleshaped** [Telmate/terraform-provider-proxmox#951]
**Tags are never-ending updated, if a list in `*.tf` config is not sorted, or uses comma instead of semicolon.**
**Reason**: Proxmox responds with sorted list, thus 2 strings never match.

```log
  # module.XYZ.proxmox_vm_qemu.vm_template will be updated in-place
      ~ tags                      = "dafdk;terraform;test" -> "dafdk;test;terraform"
```

Same issue was discussed in #771.

**Based on new user experience, 2 changes are proposed:**
(unless breaks backward-compatibility)

1. Modify README to mention correct format, https://github.com/Telmate/terraform-provider-proxmox/blob/master/docs/resources/lxc.md?plain=1#L264
   ```
   * `tags` - Tags of the container. This is only meta information.
   * `tags` - Tags of the container, semicolon-delimited. This is only meta information.
   ```
2. Sort tags before comparing state with Proxmox data in a CODE.
  1. Workaround would either manually sort the tags for each machine OR
  2. Code sorting it within TF module, e.g.

  ```
  resource "proxmox_vm_qemu" "vm_template" {
    ...
    tags        = join(";", sort(concat(["terraform"], split(";", var.tags)))) # adding "terraform" marker # workaround, as unsorted list will conflict with Proxmox's sorted state, leading to always-update situation
    ...
  }
  ```

Otherwise, every other new user of Telmate plugin will have increased learning curve and iether will re-invent a wheel, or choose to ignore the problem.

P.S. Proxmox v8.0.4 and Terraform v1.7.4, Telmate v2.9.11.

**Comment by mleone87** [comment]
the easiest way is to use lifecycle policies 

https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#ignore_changes

**Comment by needleshaped** [comment]
> the easiest way is to use lifecycle policies

Correct me if I'm wrong, but if in this case I update it manually after realising I want to add another tag, it will be also ignored, right?

In this case this more or less breaks functionality completely, apart from creation step. 
On contrary, the reason I'm writing this issue, is to hopefully have it work as it should without workarounds. Update when necessary, do not when not.

**Comment by gbrackley** [comment]
I found the same issue. I use the provider in a module, so did the following:

> tags    = join(";", sort(var.tags))

I provided the tags as an list of strings to my module. Perhaps a good 'fix' would be to have the tags attribute changed to support an list of strings. This would be a change to the interface, but the old behavior of semi-colon separated strings could be supported.

The proxmox provider should provide lossless round-tripping of the tags (so if the proxmox API or proxmox code sorts the tags then that should be allowed for).
